src/router/index.js

```diff
@@ -11,13 +11,9 @@ const routes = [
     component: Home
   },
   {
-    path: "/about",
-    name: "About",
-    // route level code-splitting
-    // this generates a separate chunk (about.[hash].js) for this route
-    // which is lazy-loaded when the route is visited.
-    component: () =>
-      import(/* webpackChunkName: "about" */ "../views/About.vue")
+    path: "/restaurant/:id",
+    name: "Restaurant",
+    component: () => import("../views/Restaurant.vue")
```

Comment:
Oh, optimization, it is! Let's dive into it then!

I'd really recommend naming your chunks. For instance, when I built your project, I got:
```
  File                                    Size          Gzipped

  dist/js/chunk-vendors.3912e405.js       285.86 KiB    94.88 KiB
  dist/js/app.a96b8788.js                 11.52 KiB     4.45 KiB
  dist/js/chunk-6fc4d1c8.b5ee5869.js      4.75 KiB      1.52 KiB
  dist/css/chunk-vendors.ca352a60.css     329.95 KiB    39.80 KiB
  dist/css/app.a4e4258e.css               0.30 KiB      0.21 KiB
  dist/css/chunk-6fc4d1c8.70a0ebdd.css    0.29 KiB      0.20 KiB
```

If you, however, name your chunk, e.g. `() => import(/* webpackChunkName: "restaurant" */ "../views/Restaurant.vue")`, you'll get much more meaningful report:

```
  File                                    Size          Gzipped

  dist/js/chunk-vendors.3912e405.js       285.86 KiB    94.88 KiB
  dist/js/app.24024211.js                 11.54 KiB     4.44 KiB
  dist/js/restaurant.999148a4.js          4.75 KiB      1.51 KiB
  dist/css/chunk-vendors.ca352a60.css     329.95 KiB    39.80 KiB
  dist/css/app.a4e4258e.css               0.30 KiB      0.21 KiB
  dist/css/restaurant.70a0ebdd.css        0.29 KiB      0.20 KiB
```

Ok, so first, let's check what happens if you remove the dynamic loading altogether.

```
  File                                    Size          Gzipped

  dist/js/chunk-vendors.3912e405.js       285.86 KiB    94.88 KiB
  dist/js/app.336f2470.js                 13.88 KiB     4.58 KiB
  dist/css/chunk-vendors.ca352a60.css     329.95 KiB    39.80 KiB
  dist/css/app.bd41a6b0.css               0.59 KiB      0.30 KiB
```

Note: `0.30 + 0.29 = 0.59`, e.g. css' size stays the same with chunking, but `11.54 + 4.75 = 16.29 > 13.88`, i.e. by chunking, you actually _increased_ the total size of your app's js code by quite much: the non-chunked version is 15% smaller, exluding vendors' chunk, which stays the same, and that is, by the way, 20 times larger than your custom code anyway.

While we're at that, you can see three other things in those numbers:

1. Your restaurant chunk is small. And by "small" I mean "really tiny". It's uncompressed js is <5 KiB and css isn't even 0.5 KiB, even combined with your main app js and css, it's still 17 KiB ,which isn't much (and remember the unchunked version is even smaller). You can benchmark the load speed by yourself, but I'd say in practice, chunks of size between 32-64 KiB are what we aim for. If you go much below that, your main page might load a little faster, but loading components dynamically will slow it down on usage (an if you go optimizing stuff, you usually end up with more than one component that could be dynamically loaded).
 
2. Compression usually works better for bigger or medium chunks rather than tiny ones (50-70% vs 20-30% here), so splitting tiny chunks degenerates compression, resulting in increase of size for the whole package.

3. Your vendors' (e.g. vuetify's) chunk is big. And by "big" I mean "really huge for a single chunk". As you can see, it's almost five the times of the optimal size of a single chunk.

Node that the size of your vendor chunk isn't as big as it could be if you haven't used vuetify's a-la-carte mode (i.e. if you imported the whole vuetify and not just the parts you actually use), then it would be larger than 1 MB. Also, bazel transpiling (you defined it in vue.config.js) makes the size go up by roughly 20-25%... which IS a big difference, as removing it would make the size go just a little below the default recommended maximal asset size, 244 KiB (js size will go down to 234.91 KiB and css size to 329.95 KiB)...

Also notice: vuetify a-la-carte mode implies that any vuetify component you use will increase the vendor chunk's size. For instance, if you remove `v-parallax`, the size will decrease by approximately 0.7% (about 1 KiB, which is more than 20% of your restaurants's chunk), but removing `v-progress-circular` would only save you 0.03%. So as you can see, you can save loading time by stripping away the used components too. There are webpack tools for visualizing how much space each dependency takes too.

That being said, it won't make your entrypoint size acceptable, as it'll still be way over 600 KiB, which is too big anyway.

If you were to optimize that, you'd need to play with `config.optimization.splitChunks`. While it can provide automatic chunk splitting (`minSize`, `maxSize`), but by default, it won't help with entrypoint being so big, as the entry point will just load all of these chunks anyway. If you really want to split your code, it would be best to split your vendor into smaller chunks and pick which component will load which vendor chunk on its own load...

...I have you already noticed just how MASSIVELY complex it gets just because you used vuetify in the first place? "Bundlephobia" isn't that paranoic, if you care for optimization. Of course, it CAN be sued in a highly optimized way, but it's nowhere near as simple as just loading your component dynamically.